### PR TITLE
Add EstablishmentPicker to DeliveryTabsInline (desktop)

### PR DESCRIPTION
**Pull request by jfplataroti**
Add `EstablishmentField` that is a textField that shows the available establishments as suggestions.

Also adds two hooks:
- `usePosition` provides the user geoLocation.
- `useEstablishments` provides a list of establishments inside a component.

Refactor theme: 
- refactored the tokens to have the same naming convention that the `BRAND_COLORS`.
- added `searchField` to global. This provides the same dimensions for the `EstablishmentField` and `AddressField` so both have the same with.

Added the `Establishment` model into services and the API call to get the establishments by coordinates (optional).

establishment-field

### CODE REVIEW

File: packages/web/components/addressField/addressField.customization.ts [hunk 1]
```diff
@@ -35,6 +35,9 @@ export const customStyles = (inputStyles, isMobile) => ({
     borderBottom: `2px solid ${theme.global.colors.gray_1}`,
     borderTopLeftRadius: '4px',
     borderTopRightRadius: '4px',
+    height: 'calc(100% - 10px)',
+    padding: '3px 12px',
```
Comment (jfplataroti) [inline packages/web/components/addressField/addressField.customization.ts:39]: this `3px` weird value was to make sure that the address field has the same height that grommet inputs.

File: packages/web/lib/theme.tsx [hunk 2]
```diff
@@ -43,19 +45,23 @@ const theme = {
       ...BRAND_COLORS,
       focus: BRAND_COLORS.brand,
     },
+    searchField: {
+      suggestionsWidth: '460px',
+      width: '320px',
+    },
```
Comment (jfplataroti) [inline packages/web/lib/theme.tsx:51]: searchField is shared between `AddressField` and `EstablishmentField`. Maybe the name `searchField` it's a bit confusing heheh. Open to suggestions :)

Comment (djoyal) [reply packages/web/lib/theme.tsx:51]: I can't think of anything better 🤷‍♂️

**Review by djoyal — COMMENTED** [review]
Looks good! There's a few small things to cleanup but on the whole it's a solid PR :)

File: packages/web/components/establishmentField/suggestionListItem.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+import storeIcon from '../../public/images/store-icon.svg';
+
+interface Props {
+  item: Establishment;
+  selected?: boolean;
+  onSelected?: (establishment: Establishment) => void;
+}
+
+const formateDistance = (distance: number) =>
```
[the remark this answers is not in the document]
Comment (jfplataroti) [reply packages/web/components/establishmentField/suggestionListItem.tsx:13]: hahah thanks!

File: packages/web/components/establishmentField/establishmentField.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  };
+
+  const onBlur = () => value && onSelected(value);
+  return (
+    <Box direction="row">
+      <TextInput
+        size="medium"
+        icon={<Image src={StoreIcon} />}
+        suggestions={suggestions}
+        onSelect={onSuggestionSelected}
```
[the remark this answers is not in the document]
Comment (jfplataroti) [reply packages/web/components/establishmentField/establishmentField.tsx:71]: oh thanks for finding that one.

File: packages/web/hooks/usePosition.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+export const usePosition = () => {
+  const [location, setLocation] = useState<Location>();
+  const [error, setError] = useState();
+
+  const onSuccess = (value: GeolocationPosition) => {
+    const { latitude: lat, longitude: long } = value.coords;
+    setLocation({ lat, long });
+  };
+
+  const onError = (err) => {
```
[the remark this answers is not in the document]
Comment (jfplataroti) [reply packages/web/hooks/usePosition.ts:19]: yeah good point 💯

File: packages/web/components/establishmentField/establishmentField.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (jfplataroti) [reply packages/web/components/establishmentField/establishmentField.tsx:71]: it looks that `onSuggestionSelect` is not in the exported typings from grommet. I asked in their help slack channel to see what's their suggestion.

File: packages/web/components/establishmentField/establishmentField.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+interface Props {
+  value?: Establishment;
+  placeholder?: string;
+  onSelected: (e: Establishment) => void;
+}
+
+const asSuggestion = (selectedEstablishment: Establishment) => (
+  item: Establishment,
+): Suggestion => ({
+  value: item.id,
```
[the remark this answers is not in the document]
Comment (jfplataroti) [reply packages/web/components/establishmentField/establishmentField.tsx:25]: great idea!

File: packages/web/hooks/useDeliveryOptions.ts [hunk 7]
```diff
@@ -66,7 +66,7 @@ export default function useDeliveryOptions() {
   function setEstablishment(establishment: Establishment): DeliveryOptions {
     const newValue = {
       ...value,
-      establishment,
+      establishment: { ...establishment },
```
[the remark this answers is not in the document]
Comment (jfplataroti) [reply packages/web/hooks/useDeliveryOptions.ts:69]: If I do that, would I be adding establishment attributes to `newValue`?

File: packages/web/components/establishmentField/establishmentField.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (jfplataroti) [reply packages/web/components/establishmentField/establishmentField.tsx]: fixed, I ended up upgrading grommet for that.

File: packages/web/hooks/useDeliveryOptions.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (djoyal) [reply packages/web/hooks/useDeliveryOptions.ts:69]: That's true - sorry, ignore my previous comment! In this case you should be able to use just `establishment`

File: packages/web/components/establishmentField/establishmentField.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (djoyal) [reply packages/web/components/establishmentField/establishmentField.tsx]: Thanks for taking care of that!

**Review by djoyal — APPROVED** [review]
Nice refactoring! Also, thanks for upgrading Grommet :)

File: packages/web/hooks/useDeliveryOptions.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (jfplataroti) [reply packages/web/hooks/useDeliveryOptions.ts:69]: Yes, sorry I never explained the scenario hehe. The reason why I'm wrapping it inside a new object,  it's to be sure that components use `establishment` in `useEffect` or `useMemo` get notified it changes.

Let me know if it makes sense.

[the remark this answers is not in the document]
Comment (djoyal) [reply packages/web/hooks/useDeliveryOptions.ts:69]: Yeah, that makes sense. Leave it as is then :)

**Pull request merged**